lib/util.js

```diff
@@ -715,42 +715,51 @@ function reduceToSingleString(output, base, braces) {
 
 // NOTE: These type checking functions intentionally don't use `instanceof`
 // because it is fragile and can be easily faked with `Object.create()`.
-exports.isArray = Array.isArray;
+exports.isArray = internalUtil.deprecate(Array.isArray,
+    'util.isArray() is deprecated. Use Array.isArray instead.');
 
 function isBoolean(arg) {
   return typeof arg === 'boolean';
 }
-exports.isBoolean = isBoolean;
+exports.isBoolean = internalUtil.deprecate(isBoolean,
+    'util.isBoolean() is deprecated. Use typeof arg === \'boolean\' instead.');
 
 function isNull(arg) {
   return arg === null;
 }
-exports.isNull = isNull;
+exports.isNull = internalUtil.deprecate(isNull,
+    'util.isNull() is deprecated. Use arg === null instead.');
 
 function isNullOrUndefined(arg) {
   return arg === null || arg === undefined;
 }
-exports.isNullOrUndefined = isNullOrUndefined;
+exports.isNullOrUndefined = internalUtil.deprecate(isNullOrUndefined,
+  'util.isNullOrUndefined() is deprecated. Use the following instead:\n' +
+  'arg === null || arg === undefined');
```

Comment:
+1 on `== null`. Last I looked at the optimizing compiler nullOrUndefined is special cased.